.buildkite/generator.jl

```diff
@@ -0,0 +1,228 @@
+const DEBUG = !haskey(ENV, "BUILDKITE")
+
+include("utils.jl")
+
+# Normally we look at the last pushed commit
+COMPARE_AGAINST="HEAD~1"
+
+# Keyword to be used in the commit message to skip a rebuild
+# TODO: Not run generator? Add to bk pipeline directly
+const SKIP_BUILD_COOKIE="[skip build]"
+
+# This variable will tell us whether we want to skip the build
+SKIP_BUILD=false
+
+const IS_PR = get(ENV, "BUILDKITE_PULL_REQUEST", "false") != "false"
+
+exec(cmd) = @assert success(pipeline(cmd, stderr=stderr, stdout=stdout))
+
+if IS_PR
+    # If we're on a PR though, we look at the entire branch at once
+    BASE_BRANCH = get(ENV, "BUILDKITE_PULL_REQUEST_BASE_BRANCH", "master")
+    @assert !isempty(BASE_BRANCH)
+
+    PR_NUMBER = parse(Int, ENV["BUILDKITE_PULL_REQUEST"])
+    TARGET_BRANCH = "remotes/origin/$(BASE_BRANCH)"
+    COMPARE_AGAINST = chomp(read(`git merge-base --fork-point $(TARGET_BRANCH) HEAD`, String))
```

Comment:
Another simplification courtesy of `readchomp`:
```suggestion
    COMPARE_AGAINST = readchomp(`git merge-base --fork-point $(TARGET_BRANCH) HEAD`)
```

Reply:
In addition to Mosè's suggestion, one way to debug why this line is currently failing would be to do a regular `run` so we can see the full log.

```suggestion
    cmd = `git merge-base --fork-point $(TARGET_BRANCH) HEAD`
    run(cmd)
    COMPARE_AGAINST = readchomp(cmd)
```

Reply:
Or, alternatively, we could only do the regular `run` if the first attempt fails. That way, we don't pollute the logs unless the command is failing.

```suggestion
    cmd = `git merge-base --fork-point $(TARGET_BRANCH) HEAD`
    run(cmd)
    COMPARE_AGAINST = try
        readchomp(cmd)
    catch
        run(cmd)
        rethrow()
    end
```